MIG-1531: MTC docs missing Direct Migration Requirements (#73269)
openshift/openshift-docs: migration_toolkit_for_containers/mtc-direct-migration-requirements.adoc

:_mod-docs-content-type: ASSEMBLY
[id="mtc-direct-migration-requirements"]
= Direct Migration Requirements
include::_attributes/common-attributes.adoc[]
include::_attributes/attributes-openshift-dedicated.adoc[]
:context: mtc-direct-migration-requirements
:mtc-direct-migration-requirements:

toc::[]

Direct Migration is available from {mtc-full} ({mtc-short}) 1.4.0. 

There are two parts of the Direct Migration:

* Direct Volume Migration 

* Direct Image Migration 

Direct Migration enables the migration of persistent volumes and internal images directly from the source cluster to the destination cluster without an intermediary replication repository (object storage) 

> == Prerequisites
> 
> * Expose the internal registries for both clusters (source and destination) involved in the migration for external traffic.
> 
> * Ensure the remote source and destination clusters are able to communicate using {OCP} routes on port 443.

Author: ```suggestion
== Prerequisites

* Expose the internal registries for both clusters (source and destination) involved in the migration for external traffic.

* Ensure the remote source and destination clusters can communicate using {OCP} routes on port 443.

* Configure the exposed registry route in the source and destination {mtc-short} clusters; do this by specifying the `spec.exposedRegistryPath` field or from the {mtc-short} UI. 
+
[NOTE] 
==== 
* If the destination cluster is the same as the host cluster (where a migration controller exists), there is no need to configure the exposed registry route for that particular {mtc-short} cluster.

* The `spec.exposedRegistryPath` is only required for Direct Image Migration and not Direct Volume Migration.
====

* The two spec flags in `MigPlan` custom resource (CR) `indirectImageMigration` and `indirectVolumeMigration` need to be set to `false` for Direct Migration to be executed. The default value for these flags is `false`.

The Direct Migration feature of {mtc-short} uses the `rsync` utility.
```

:mtc-direct-migration-requirements:

toc::[]

Direct Migration is available from {mtc-full} ({mtc-short}) 1.4.0. 

There are two parts of the Direct Migration:

* Direct Volume Migration 

* Direct Image Migration 

Direct Migration enables the migration of persistent volumes and internal images directly from the source cluster to the destination cluster without an intermediary replication repository (object storage) 

== Prerequisites

* Expose the internal registries for both clusters (source and destination) involved in the migration for external traffic.

* Ensure the remote source and destination clusters can communicate using {OCP} routes on port 443.

* Configure the exposed registry route in the source and destination {mtc-short} clusters; do this by specifying the `spec.exposedRegistryPath` field or from the {mtc-short} UI. 
+
[NOTE] 
==== 
* If the destination cluster is the same as the host cluster (where a migration controller exists), there is no need to configure the exposed registry route for that particular {mtc-short} cluster.

* The `spec.exposedRegistryPath` is only required for Direct Image Migration and not Direct Volume Migration.
====

> * The two spec flags in `MigPlan` custom resource (CR) `indirectImageMigration` and `indirectVolumeMigration` need to be set to `false` for Direct Migration to be executed. The default value for these flags is `false`.

Author: ```suggestion
* Ensure the two spec flags in `MigPlan` custom resource (CR) `indirectImageMigration` and `indirectVolumeMigration` are set to `false` for Direct Migration to be performed. The default value for these flags is `false`.

```

:_mod-docs-content-type: ASSEMBLY
[id="mtc-direct-migration-requirements"]
= Direct Migration Requirements
include::_attributes/common-attributes.adoc[]
:context: mtc-direct-migration-requirements

> :mtc-direct-migration-requirements:

Author: ```suggestion
:mtc-direct-migration-requirements:

toc::[]
```

:_mod-docs-content-type: ASSEMBLY
[id="mtc-direct-migration-requirements"]
= Direct Migration Requirements

> include::_attributes/common-attributes.adoc[]

Author: ```suggestion
include::_attributes/common-attributes.adoc[]
include::_attributes/attributes-openshift-dedicated.adoc[]
```

:_mod-docs-content-type: ASSEMBLY
[id="mtc-direct-migration-requirements"]
= Direct Migration Requirements
include::_attributes/common-attributes.adoc[]
include::_attributes/attributes-openshift-dedicated.adoc[]
:context: mtc-direct-migration-requirements
:mtc-direct-migration-requirements:

> toc::[]

Author: ```suggestion
toc::[]

Direct Migration is available from {mtc-full} ({mtc-short}) 1.4.0. 

There are two parts of the Direct Migration:

* Direct Volume Migration 

* Direct Image Migration 

Direct Migration enables the migration of persistent volumes and internal images directly from the source cluster to the destination cluster without an intermediary replication repository (object storage) 

== Prerequisites

* Expose the internal registries for both clusters (source and destination) involved in the migration for external traffic.

* Ensure the remote source and destination clusters are able to communicate using {OCP} routes on port 443.
```
